### PR TITLE
[PWA] Add Playwright for e2e tests

### PR DESCRIPTION
**Pull request by tervay**
There are still some improvements to make here but this is a good first step.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -265,6 +265,35 @@ jobs:
...
+  pwa-e2e:
+    name: "[pwa] E2E Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Create dovenv
+        run: ./ops/pwa/create_dotenv.sh --env
+        env:
+          VITE_TBA_API_READ_KEY: ${{ TBA_API_READ_KEY }}
```
Comment (fangeugene) [inline .github/workflows/pull_request.yml:277]:
```suggestion
          VITE_TBA_API_READ_KEY: ${{ secrets.TBA_API_READ_KEY }}
```

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -265,6 +265,37 @@ jobs:
           TBA_API_READ_KEY: TEST_KEY
       - name: Run Build
         run: npm --prefix pwa run build
+  pwa-e2e:
+    name: "[pwa] E2E Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Create dovenv
```
Comment (fangeugene) [inline .github/workflows/pull_request.yml:274]: Is this still necessary with the env var specified down below?

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -265,6 +265,37 @@ jobs:
...
+          node-version: "20.16.0"
+          cache: "npm"
+
+      - name: Install Dependencies
+        working-directory: pwa
+        run: npm ci
+      
+      - name: Install Playwright Browsers
+        working-directory: pwa
+        run: npx playwright install --with-deps
```
Comment (fangeugene) [inline .github/workflows/pull_request.yml:291]: Why are these using npx instead of npm, when we're installing the dev dependency for playwright?

Comment (tervay) [reply .github/workflows/pull_request.yml:291]: Dunno - just followed docs: https://playwright.dev/docs/intro

File: pwa/app/components/tba/nav.tsx [hunk 4]
```diff
@@ -105,7 +105,11 @@ export const Nav = () => {
     <div className="fixed z-10 flex w-full grow justify-center bg-primary shadow-md">
       <GlobalLoadingProgress />
       <NavigationMenu className="gap-6 px-4 py-2.5">
-        <Link to="/" className="flex items-center gap-3 hover:no-underline">
+        <Link
+          to="/"
+          className="flex items-center gap-3 hover:no-underline"
+          id="tba-logo-link"
```
Comment (fangeugene) [inline pwa/app/components/tba/nav.tsx:111]: is this change necessary? The tests don't seem to reference this

File: .github/workflows/pull_request.yml (hunk 3 above)
Comment (fangeugene) [reply .github/workflows/pull_request.yml:291]: ok 🤷

**Review by fangeugene — DISMISSED** [review]
